src/geo/leaflet/leaflet-cartodb-vector-layer-group-view.js

```diff
@@ -27,6 +27,17 @@ var LeafletCartoDBVectorLayerGroupView = CartoDBd3Layer.extend({
       this.setCartoCSS(index, style);
     }, this);
 
+    layerModel.layers.bind('change:meta', function (child, meta) {
+      if (this.model.get('type') === 'namedmap') {
+        var index;
+        child.collection.models.filter(function (c, i) { if (child.cid === c.cid) index = i; });
+        this.options.styles[index - 1] = meta.cartocss;
+        if (this.options.styles.filter(function (p) { return typeof p === 'undefined'; }).length === 0) {
```

Comment:
This conditional is really hard to understand for me. We could extract one or two variables with good names to make it a bit easier. Or even extract a method with that check:

```
// ...
if (this._somethingIsTrue()) {
  // ...
}
// ...